src/test/java/uk/gov/dhsc/htbhf/page/GuidancePage.java

```diff
@@ -3,55 +3,80 @@
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
-import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
 
-import java.util.Map;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
 
 /**
  * Page object for the guidance pages of the application
  */
-public class GuidancePage {
+public class GuidancePage extends BasePage {
+
+    public static final String APPLY_PAGE = "Apply for Healthy Start";
 
-    private static final String APPLY_PAGE = "Apply for Healthy Start";
+    private GuidancePageMetadata currentPage;
+
+    public GuidancePage(WebDriver webDriver, String baseUrl, WebDriverWait wait, String pageName) {
+        super(webDriver, baseUrl, wait);
+        this.currentPage = GuidancePageMetadata.findByName(pageName);
+    }
 
-    //TODO MRS 2019-08-07: Guidance page enum? Look into this when guidance-pages.feature is done
-    private static final Map<String, String> GUIDANCE_PAGES = Map.of(
-            "How it works", "/",
-            "Eligibility", "/eligibility",
-            "What you’ll get", "/what-you-get",
-            "What you can buy", "/buy",
-            "Using your card", "/using-your-card",
-            APPLY_PAGE, "/apply",
-            "Report a change", "/report-a-change"
-    );
+    public static GuidancePage buildApplyGuidancePage(WebDriver webDriver, String baseUrl, WebDriverWait wait) {
+        return new GuidancePage(webDriver, baseUrl, wait, APPLY_PAGE);
+    }
 
-    private WebDriver webDriver;
-    private String baseUrl;
-    private WebDriverWait wait;
+    @Override
+    String getPath() {
+        return currentPage.getPagePath();
+    }
 
-    public GuidancePage(WebDriver webDriver, String baseUrl, WebDriverWait wait) {
-        this.webDriver = webDriver;
-        this.baseUrl = baseUrl;
-        this.wait = wait;
+    @Override
+    String getPageName() {
+        return currentPage.getPageName();
     }
 
-    public void openGuidancePage(String pageName) {
-        String path = GUIDANCE_PAGES.get(pageName);
-        webDriver.get(baseUrl + path);
-        wait.until(ExpectedConditions.titleIs(getPageTitleForPath(pageName)));
+    @Override
+    String getPageTitle() {
+        return "GOV.UK - " + currentPage.getPageName();
     }
 
     public void clickStartButton() {
         WebElement startButton = webDriver.findElement(By.className("govuk-button--start"));
         startButton.click();
     }
 
-    public void openApplyPage() {
-        openGuidancePage(APPLY_PAGE);
+    public boolean hasPreviousLink() {
```

Comment:
This method is more `shouldHavePreviousLink`, isn't it? It's not actually testing whether such a link exists

Reply:
Yes, that's right, I'll rename.